cmd/syncthing/main.go

```diff
@@ -269,20 +284,20 @@ func main() {
 		osutil.HideConsole()
 	}
 
-	if options.confDir != "" {
-		// Not set as default above because the string can be really long.
-		if !filepath.IsAbs(options.confDir) {
-			var err error
-			options.confDir, err = filepath.Abs(options.confDir)
-			if err != nil {
-				l.Warnln("Failed to make options path absolute:", err)
-				os.Exit(syncthing.ExitError.AsInt())
-			}
-		}
-		if err := locations.SetBaseDir(locations.ConfigBaseDir, options.confDir); err != nil {
-			l.Warnln(err)
-			os.Exit(syncthing.ExitError.AsInt())
+	// Not set as default above because the string can be really long.
+	var err error
+	if options.homeDir != "" {
+		if err = setLocation(options.homeDir, locations.ConfigBaseDir); err == nil {
+			err = setLocation(options.homeDir, locations.DataBaseDir)
 		}
+	} else if options.confDir != "" {
+		err = setLocation(options.confDir, locations.ConfigBaseDir)
+	} else if options.dataDir != "" {
```

Comment:
Only check -data if I didn't give -config or -home? It feels to me these should take effect in order, first -home and then -config/-data independently